### ISSUE TITLE
Error 'ValueError: Cannot use 'filtered_detections/map/while/strided_slice_1' as input to 'filtered_detections/map/while/ones/packed' because 'filtered_detections/map/while/strided_slice_1' is in a while loop. See info log for more details.'

### ISSUE THREAD

**Issue by Samjith888** [fizyr/keras-retinanet#1118]
```
(base) C:\Users\Samjith.CP\Desktop\tdetect>retinanet-train --weights resnet50_coco_best_v2.1.0.h5 --batch-size 4 --steps 67 --epochs 20 --snapshot-path snapshots --tensorboard-dir tensorboard csv dataset/train.csv dataset/classes.csv
Using TensorFlow backend.
WARNING:tensorflow:From c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\bin\train.py:65: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.

WARNING:tensorflow:From c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\bin\train.py:67: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

2019-09-22 11:39:37.861009: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
Creating model, this may take a second...
WARNING:tensorflow:From c:\users\samjith.cp\appdata\roaming\python\python37\site-packages\keras\backend\tensorflow_backend.py:4070: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.

WARNING:tensorflow:From c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\backend\tensorflow_backend.py:62: The name tf.image.resize_images is deprecated. Please use tf.image.resize instead.

c:\users\samjith.cp\appdata\roaming\python\python37\site-packages\keras\engine\saving.py:1316: UserWarning: Skipping loading of weights for layer classification_submodel due to mismatch in shape ((3, 3, 256, 9) vs (720, 256, 3, 3)).
  weight_values[i].shape))
c:\users\samjith.cp\appdata\roaming\python\python37\site-packages\keras\engine\saving.py:1316: UserWarning: Skipping loading of weights for layer classification_submodel due to mismatch in shape ((9,) vs (720,)).
  weight_values[i].shape))
tracking <tf.Variable 'Variable:0' shape=(9, 4) dtype=float32> anchors
tracking <tf.Variable 'Variable_1:0' shape=(9, 4) dtype=float32> anchors
tracking <tf.Variable 'Variable_2:0' shape=(9, 4) dtype=float32> anchors
tracking <tf.Variable 'Variable_3:0' shape=(9, 4) dtype=float32> anchors
tracking <tf.Variable 'Variable_4:0' shape=(9, 4) dtype=float32> anchors
Traceback (most recent call last):
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 2084, in ones
    output = _constant_if_small(one, shape, dtype, name)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 1828, in _constant_if_small
    if np.prod(shape) < 1000:
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\gen_math_ops.py", line 5521, in less
    "Less", x=x, y=y, name=name)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\util\deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 3616, in create_op
    op_def=op_def)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 2043, in __init__
    self._control_flow_post_processing()
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 2052, in _control_flow_post_processing
    control_flow_util.CheckInputFromValidContext(self, input_tensor.op)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_util.py", line 350, in CheckInputFromValidContext
    raise ValueError(error_msg + " See info log for more details.")
ValueError: Cannot use 'filtered_detections/map/while/strided_slice_1' as input to 'filtered_detections/map/while/ones/Less' because 'filtered_detections/map/while/strided_slice_1' is in a while loop. See info log for more details.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Samjith.CP\Anaconda3\Scripts\retinanet-train-script.py", line 11, in <module>
    load_entry_point('keras-retinanet==0.4.1', 'console_scripts', 'retinanet-train')()
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\bin\train.py", line 440, in main
    freeze_backbone=args.freeze_backbone
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\bin\train.py", line 112, in create_models
    prediction_model = retinanet_bbox(model=model)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\models\retinanet.py", line 366, in retinanet_bbox
    )([boxes, classification] + other)
  File "c:\users\samjith.cp\appdata\roaming\python\python37\site-packages\keras\engine\base_layer.py", line 489, in __call__
    output = self.call(inputs, **kwargs)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\layers\filter_detections.py", line 179, in call
    parallel_iterations=self.parallel_iterations
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\backend\tensorflow_backend.py", line 29, in map_fn
    return tensorflow.map_fn(*args, **kwargs)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\map_fn.py", line 268, in map_fn
    maximum_iterations=n)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_ops.py", line 3501, in while_loop
    return_same_structure)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_ops.py", line 3012, in BuildLoop
    pred, body, original_loop_vars, loop_vars, shape_invariants)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_ops.py", line 2937, in _BuildLoop
    body_result = body(*packed_vars_for_body)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_ops.py", line 3456, in <lambda>
    body = lambda i, lv: (i + 1, orig_body(*lv))
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\map_fn.py", line 257, in compute
    packed_fn_values = fn(packed_values)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\layers\filter_detections.py", line 171, in _filter_detections
    nms_threshold         = self.nms_threshold,
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\keras_retinanet-0.4.1-py3.7-win-amd64.egg\keras_retinanet\layers\filter_detections.py", line 76, in filter_detections
    labels = c * keras.backend.ones((keras.backend.shape(scores)[0],), dtype='int64')
  File "c:\users\samjith.cp\appdata\roaming\python\python37\site-packages\keras\backend\tensorflow_backend.py", line 996, in ones
    v = tf.ones(shape=shape, dtype=dtype, name=name)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 2093, in ones
    shape = ops.convert_to_tensor(shape, dtype=dtypes.int32)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1087, in convert_to_tensor
    return convert_to_tensor_v2(value, dtype, preferred_dtype, name)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1145, in convert_to_tensor_v2
    as_ref=False)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1224, in internal_convert_to_tensor
    ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 1145, in _autopacking_conversion_function
    return _autopacking_helper(v, dtype, name or "packed")
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 1095, in _autopacking_helper
    return gen_array_ops.pack(elems_as_tensors, name=scope)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\gen_array_ops.py", line 6971, in pack
    "Pack", values=values, axis=axis, name=name)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\util\deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 3616, in create_op
    op_def=op_def)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 2043, in __init__
    self._control_flow_post_processing()
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 2052, in _control_flow_post_processing
    control_flow_util.CheckInputFromValidContext(self, input_tensor.op)
  File "c:\users\samjith.cp\anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_util.py", line 350, in CheckInputFromValidContext
    raise ValueError(error_msg + " See info log for more details.")
ValueError: Cannot use 'filtered_detections/map/while/strided_slice_1' as input to 'filtered_detections/map/while/ones/packed' because 'filtered_detections/map/while/strided_slice_1' is in a while loop. See info log for more details.
```

OS : windows
Python 3.7
Tensorflow -cpu

**Comment by hgaiser** [comment]
What version of tensorflow are you using?

**Comment by Samjith888** [comment]
1.14.0

**Comment by yustiks** [comment]
I have the same issue

**Comment by hgaiser** [comment]
> I have the same issue

The issue comes from [here](https://github.com/tensorflow/tensorflow/issues/33787), which is also ported to keras 2.3.1. If you want a working version until that issue is resolved, you can use tensorflow 1.14, keras 2.2.5 and keras-retinanet 0.5.1.
